app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -298,22 +302,34 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
     }
 
     override fun navToQuickSettingsSheet(tab: SessionState, sitePermissions: SitePermissions?) {
-        requireComponents.useCases.trackingProtectionUseCases.containsException(tab.id) { contains ->
-            runIfFragmentIsAttached {
-                val isTrackingProtectionEnabled = tab.trackingProtection.enabled && !contains
-                val directions =
-                    BrowserFragmentDirections.actionBrowserFragmentToQuickSettingsSheetDialogFragment(
-                        sessionId = tab.id,
-                        url = tab.content.url,
-                        title = tab.content.title,
-                        isSecured = tab.content.securityInfo.secure,
-                        sitePermissions = sitePermissions,
-                        gravity = getAppropriateLayoutGravity(),
-                        certificateName = tab.content.securityInfo.issuer,
-                        permissionHighlights = tab.content.permissionHighlights,
-                        isTrackingProtectionEnabled = isTrackingProtectionEnabled,
+        val useCase = requireComponents.useCases.trackingProtectionUseCases
+        useCase.containsException(tab.id) { hasTrackingProtectionException ->
+            lifecycleScope.launch(Dispatchers.Main) {
+                val cookieBannersStorage = requireComponents.core.cookieBannersStorage
+                val hasCookieBannerException = withContext(Dispatchers.IO) {
+                    cookieBannersStorage.hasException(
```

Comment:
I was curious about this method hasException(uri: String, privateBrowsing: Boolean): Boolean should not be better to call it only from QuickSettingsController . 
Because it is used only in QuickSettingsSheetDialogFragment and it passed by bundle from TrackingProtectionPanelInteractor, DefaultCookieBannerDetailsController. ,ExternalAppBrowserFragment ,BrowserFragment and ConnectionDetailsController.

Reply:
It's to always has the latest state of the value in the storage, as the value could be changed in the inner panels, if we passed as nav argument, the value could be wrong, and not reflecting the state in the storage.

Reply:
But the value is passed as nav argument .

Reply:
In Focus this method hasException I called it only one time .

Reply:
As Focus and Fenix has different architecture than Focus could be easier to implement than in Focus   .

Reply:
Just to give an idea of the dialogs hierarchy. 

## Parents
### (Normal/Private browsing) BrowserFragment

* QuickSettingsSheetDialogFragment 
    * Dialog Connection
      * Dialog Connection Details
    * Dialog Tracking Protection 
      * Dialog Tracking Protection details
        * Dialog Tracking Protection inner details 
    * Dialog Cookie Banner Details


###  (Custom Tabs, and PWAs) ExternalAppBrowserFragment

* QuickSettingsSheetDialogFragment 
    * Dialog Connection
      * Dialog Connection Details
    * Dialog Tracking Protection 
      * Dialog Tracking Protection details
        * Dialog Tracking Protection inner details 
    * Dialog Cookie Banner Details

Reply:
From any inner dialogs users can go back, and change the value.